tests/end-to-end/director/labeling_api_test.go

```diff
@@ -0,0 +1,845 @@
+	getApp := gcli.NewRequest(
+		fmt.Sprintf(`query {
+				result: application(id: %s,) {
+						%s
+					}
+				}`,
+			firstApp.ID, tc.gqlFieldsProvider.ForApplication()))
+
+	app := graphql.Application{}
+	// WHEN
+	err = tc.RunQuery(ctx, getApp, app)
+
+	//THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, app)
+	assert.NotContains(t, app.Labels, labelKey)
+
+	// Check if LabelDefinition was deleted
+	// GIVEN
+	getLabelDefinitionRequest := gcli.NewRequest(
+		fmt.Sprintf(`query {
+				result: labelDefinition(key: %s) {
+						%s
+					}
+				}`,
+			labelKey, tc.gqlFieldsProvider.ForLabelDefinition()))
+
+	// WHEN
+	err = tc.RunQuery(ctx, getLabelDefinitionRequest, nil)
+
+	//THEN
+	require.Error(t, err)
+}
+
+func TestDeleteScenarioLabel(t *testing.T) {
+	//Create application
+	// GIVEN
+	ctx := context.Background()
+	in := generateSampleApplicationInput("first")
+	appInputGQL, err := tc.graphqlizer.ApplicationInputToGQL(in)
+	require.NoError(t, err)
+	createReq := gcli.NewRequest(
+		fmt.Sprintf(`mutation {
+  				result: createApplication(in: %s) {
+    					%s
+					}
+				}`, appInputGQL, tc.gqlFieldsProvider.ForApplication()))
+	firstApp := graphql.Application{}
+
+	//WHEN
+	err = tc.RunQuery(ctx, createReq, &firstApp)
+
+	//THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, firstApp.ID)
+	defer deleteApplication(t, firstApp.ID)
+
+	//Try to delete scenario label on application
+	// GIVEN
+	labelKey := "scenario"
+	deleteAppLabelRequest := gcli.NewRequest(
+		fmt.Sprintf(`mutation {
+				result: deleteApplicationLabel(applicationID: %s, key: %s) {
+						%s
+					}
+				}`,
+			firstApp.ID, labelKey, tc.gqlFieldsProvider.ForApplicationLabel()))
+
+	// WHEN
+	err = tc.RunQuery(ctx, deleteAppLabelRequest, nil)
+
+	//THEN
+	require.Error(t, err)
+}
+
+func TestDeleteDefaultValueInScenarioLabel(t *testing.T) {
+	//Create application
+	// GIVEN
+	ctx := context.Background()
+	in := generateSampleApplicationInput("first")
+	appInputGQL, err := tc.graphqlizer.ApplicationInputToGQL(in)
+	require.NoError(t, err)
+	createReq := gcli.NewRequest(
+		fmt.Sprintf(`mutation {
+  				result: createApplication(in: %s) {
+    					%s
+					}
+				}`, appInputGQL, tc.gqlFieldsProvider.ForApplication()))
+	firstApp := graphql.Application{}
+
+	//WHEN
+	err = tc.RunQuery(ctx, createReq, &firstApp)
+
+	//THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, firstApp.ID)
+	defer deleteApplication(t, firstApp.ID)
+
+	//Try to delete scenario label on application
+	// GIVEN
+	labelKey := "scenario"
+	labelValue := "notDefault"
+	deleteAppLabelRequest := gcli.NewRequest(
+		fmt.Sprintf(`mutation {
+				result: setApplicationLabel(applicationID: %s, key: %s, value: %s) {
+						%s
+					}
+				}`,
+			firstApp.ID, labelKey, labelValue, tc.gqlFieldsProvider.ForApplicationLabel()))
+
+	// WHEN
+	err = tc.RunQuery(ctx, deleteAppLabelRequest, nil)
+
+	//THEN
+	require.Error(t, err)
+}
+
+func TestSearchByLabels(t *testing.T) {
```

Comment:
Maybe some loops/helper functions could help shorten this test?

Reply:
Yes, please refactor it

Reply:
moved common parts to `fixtures_queries_test` and `fixtures_request_test` files